### DISCUSSION TITLE
Get model components with current parameters guess

### DISCUSSION THREAD

**Discussion by DSimonne** [lmfit/lmfit-py#783] [Q&A]
Hello

Is it possible to get the values over a range x of the components of a model, **before fitting**, with a set of guess parameters ?
I try to explain:
When I create a model, I can get an initial guess by typing:
```python
init = mod.eval(pars, x=x)
```

But to get the components values, I can only type: 
```python
components = mod.eval_components(x=x)
```

without giving parameters. And the resulting components are computed with the basic parameters for each component (e.g. amplitude = 1, etc ...).

Of course then I can fit the model and get the components with:
```python
out = mod.fit()
components = out.eval_components(x=x)
```

But I would like to plot my initial guess before fitting to see how accurate it is.

Thank you for the help, I hope it is clear !

David

**Comment by newville** [comment]
@DSimonne yeah, I guess that `CompositeModel.eval_components` should have and use a `params` keyword arg.

A PR would be welcome.

**Reply by DSimonne** [reply]
Hi @newville,

I'll try and work on that !
